Src/FluentAssertions/NumericAssertionsExtensions.cs

```diff
@@ -9,6 +9,676 @@ namespace FluentAssertions
+        /// <param name="nearbyValue">
+        /// The nearby value to compare the actual value with.
+        /// </param>
+        /// <param name="delta">
+        /// The maximum amount of which the two values may differ.
+        /// </param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])"/> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because"/>.
+        /// </param>
+        public static AndConstraint<NumericAssertions<sbyte>> BeCloseTo(this NumericAssertions<sbyte> parent,
+            sbyte nearbyValue, byte delta, string because = "",
+            params object[] becauseArgs)
+        {
+            sbyte actualValue = (sbyte)parent.Subject;
+            sbyte minValue = (sbyte)(nearbyValue - delta);
+            if (minValue > nearbyValue)
+            {
+                minValue = sbyte.MinValue;
+            }
+
+            sbyte maxValue = (sbyte)(nearbyValue + delta);
+            if (maxValue < nearbyValue)
+            {
+                maxValue = sbyte.MaxValue;
+            }
+
+            Execute.Assertion
+                .ForCondition(minValue <= actualValue && actualValue <= maxValue)
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected {context:value} to be within {0} from {1}{reason}, but found {2}.",
+                    delta, nearbyValue, actualValue);
+
+            return new AndConstraint<NumericAssertions<sbyte>>(parent);
+        }
+
+        /// <summary>
+        /// Asserts an integral value is within another value by a specified value.
+        /// </summary>
+        /// <param name="parent">The <see cref="NumericAssertions{T}"/> object that is being extended.</param>
+        /// <param name="nearbyValue">
+        /// The nearby value to compare the actual value with.
+        /// </param>
+        /// <param name="delta">
+        /// The maximum amount of which the two values may differ.
+        /// </param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])"/> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because"/>.
+        /// </param>
+        public static AndConstraint<NumericAssertions<byte>> BeCloseTo(this NumericAssertions<byte> parent,
+            byte nearbyValue, byte delta, string because = "",
+            params object[] becauseArgs)
+        {
+            byte actualValue = (byte)parent.Subject;
```

Comment:
I wonder if you can somehow replace this all this repetition with a mathod that takes an `IComparable`